env/env.go

```diff
@@ -33,25 +33,28 @@ func LoadEnv(config interface{}) error {
 		return err
 	}
 
-	// Format .env file keys.
-	for k, v := range dotEnvMap {
-		delete(dotEnvMap, k)
-		if strings.HasPrefix(k, goconfig.PrefixEnv) {
-			k = strings.TrimPrefix(k, goconfig.PrefixEnv+"_")
-		}
-		dotEnvMap[strings.ToLower(k)] = v
-	}
-
 	configType := reflect.TypeOf(config).Elem()
 	configValue := reflect.ValueOf(config).Elem()
 
 	for i := 0; i < configType.NumField(); i++ {
 		field := configType.Field(i)
-		tag := field.Tag.Get("cfg")
-		if tag == "" {
+
+		confKey := field.Tag.Get("env")
+		if confKey == "" {
+			confKey = field.Tag.Get("cfg")
+		}
+		if confKey == "-" {
 			continue
 		}
-		value, ok := dotEnvMap[tag]
+		if confKey == "" {
+			confKey = strings.ToUpper(field.Name)
+		}
+
+		prefix := ""
+		if goconfig.PrefixEnv != "" {
+			prefix = goconfig.PrefixEnv + "_"
+		}
+		value, ok := dotEnvMap[prefix+confKey]
```

Comment:
IIUC then if `goconfig.PrefixEnv` is let's say `MY_REFIX` and I have a struct with field ```Foo `env:"FOO"` ``` it would read the variable from `MY_PREFIX_FOO`, correct? I'm not sure I like that, it is a bit surprising. I expect the value of the `env` tag to be "absolute".

Reply:
> IIUC then if goconfig.PrefixEnv is let's say MY_REFIX and I have a struct with field Foo `env:"FOO"`  it would read the variable from MY_PREFIX_FOO, correct?

Correct.

> I'm not sure I like that, it is a bit surprising. I expect the value of the env tag to be "absolute".

Fair point. On the other hand, it's reasonable to assume that `.env` format expects the very same format as env variables in which case coupling it together with `goconfig.PrefixEnv` makes sense IMO. Thoughts?

Reply:
I understand your reasoning, but I agree with Michal. I'd expect the field names to behave like the other types.

I'm not even sure what a generated `ENV` would look like with `PrefixEnv` (considering your approach). That would be two times `PrefixEnv`, right? One for the `.env` file fields and the other for `ENV` variables derived from `.env` file fields.

Reply:
If the `ENV` var is generated from `cfg` entry then yes and I have no concerns.

Reply:
Correct. `env:` tag impacts `.env` entries only. Regular `ENV` vars still adhere to `cfg:`.

Reply:
Side note: Already removed the `PrefixEnv` dependency from `.env` plugin. If prefixes are needed in `.env` entries, they need to be explicitly specified via `env` tag.